### PR TITLE
chore(e2e): Add E2E tests for new My Queries use cases COMPASS-8152

### PR DESCRIPTION
**Pull request by lerouxb**
* when you have multiple active connections that have the namespace
* when you have multiple active connections, and only one of them has the namespace
* when you have multiple active connections, and none of them have the namespace

### CODE REVIEW

File: packages/compass-components/src/components/workspace-tabs/tab.tsx [hunk 1]
```diff
@@ -254,7 +254,7 @@ function Tab({
           tabIndex={isSelected ? 0 : -1}
           aria-controls={tabContentId}
           data-testid="workspace-tab-button"
-          data-connectionName={connectionName}
+          data-connection-name={connectionName}
```
Comment (lerouxb) [inline packages/compass-components/src/components/workspace-tabs/tab.tsx:257]: small drive-by: we use data-connection-name already in the sidebar, so I changed this to match because it threw me off at first.

File: packages/compass-e2e-tests/tests/search-indexes.test.ts [hunk 2]
```diff
@@ -203,6 +203,8 @@ describe('Search Indexes', function () {
   });
 
   afterEach(async function () {
+    await screenshotIfFailed(compass, this.currentTest);
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/search-indexes.test.ts:206]: Drive-by: One of the new search indexes tests sometimes flakes and I noticed that it disconnected BEFORE we take the error screenshot and then obviously the screenshot is useless. This is something I broke probably in the previous e2e PR and just didn't notice.

File: packages/compass-e2e-tests/tests/my-queries-tab.test.ts [hunk 3]
```diff
@@ -40,282 +45,333 @@ async function openMenuForQueryItem(
...
+  const connectionStrings = [DEFAULT_CONNECTION_STRING_1];
+  if (TEST_MULTIPLE_CONNECTIONS) {
+    connectionStrings.push(DEFAULT_CONNECTION_STRING_2);
+  }
+  clients = connectionStrings.map(
+    (connectionString) => new MongoClient(connectionString)
+  );
+
+  await Promise.all(clients.map((client) => client.connect()));
+  */
```
Comment (paula-stacho) [inline packages/compass-e2e-tests/tests/my-queries-tab.test.ts:171]: leftover?

File: packages/compass-e2e-tests/tests/my-queries-tab.test.ts [hunk 4]
```diff
@@ -40,282 +45,333 @@ async function openMenuForQueryItem(
...
+    }
+  );
 
   context(
-    'when a user has a saved query associated with a collection that does not exist',
+    'when a user has a saved query associated with a collection that does not exist in the single connection',
     function () {
-      const favoriteQueryName = 'list of numbers greater than 10 - query';
-      const newCollectionName = 'numbers-renamed';
+      it('opens a model where users can permanently associate a new namespace for an aggregation/query', async function () {
```
Comment (paula-stacho) [inline packages/compass-e2e-tests/tests/my-queries-tab.test.ts:359]: *modal

File: packages/compass-e2e-tests/tests/my-queries-tab.test.ts [hunk 5]
```diff
@@ -404,4 +453,200 @@ describe('My Queries tab', function () {
...
+          return activeConnectionName === DEFAULT_CONNECTION_NAME_2;
+        });
+      });
+    }
+  );
+
+  context(
+    'when a user has multiple connections and none of them contain the namespace',
+    function () {
+      it('opens a model where users can select a connection and permanently associate a new namespace for an aggregation/query', async function () {
```
Comment (paula-stacho) [inline packages/compass-e2e-tests/tests/my-queries-tab.test.ts:509]: *modal

File: packages/compass-e2e-tests/tests/my-queries-tab.test.ts (hunk 3 above)
Comment (lerouxb) [reply packages/compass-e2e-tests/tests/my-queries-tab.test.ts:171]: ugh now I merged before spotting this. Will remove in the next PR that adds E2E tests.
